lib/media/segment_prefetch.js

```diff
@@ -227,10 +230,13 @@ shaka.media.SegmentPrefetchOperation = class {
    * @public
    */
   dispatchFetch(reference, stream) {
+    let remaining = new Uint8Array(0);
     this.operation_ = this.fetchDispatcher_(
         reference, stream, async (data) => {
+          remaining = shaka.util.Uint8ArrayUtils.concat(remaining, data);
           if (this.streamDataCallback_) {
-            await this.streamDataCallback_(data);
+            await this.streamDataCallback_(remaining);
+            remaining = new Uint8Array(0);
```

Comment:
I'm not super familiar with the lifecycle of this system.
Is there any situation where `fetchDispatcher_` could get all of the data before `this.streamDataCallback_` is set?
If there is, all of the fetched data would still disappear in that case.

Reply:
If the request has finished, all the data is saved but not here, but in SegmentPrefetch, so no data is lost.